### DISCUSSION TITLE
An input value changes - I want useQuery to refetch the endpoint with the new value

### DISCUSSION THREAD

**Discussion by efreethy** [trpc/trpc#4520] [General]
The question is pretty simply stated, yet its still unclear to me how this is achieved. The typings for useQuery don't make this easier to reason about. And the API docs reference tanstack - however tanstack does this much differently

**Comment by Nick-Lucas** [comment]
If you change the input state, useQuery will refetch (in Tanstack you change the query key, but we calculate the query key for you)

You can also invalidate queries as normal after mutations, just using the trpc.useContext hook to get that method

**Reply by efreethy** [reply]
Thank you 🙏
